nilearn/_utils/cache_mixin.py

```diff
@@ -208,17 +208,45 @@ def _cache(self, func, func_memory_level=1, **kwargs):
         if not hasattr(self, "memory"):
             self.memory = Memory(cachedir=None, verbose=verbose)
         if isinstance(self.memory, _basestring):
-            self.memory = Memory(cachedir=self.memory, verbose=verbose)
+            cache_dir = self.memory
+            if nilearn.EXPAND_PATH_WILDCARDS:
+                cache_dir = os.path.expanduser(cache_dir)
+
+            # Perform some verifications on given path.
+            split_cache_dir = os.path.split(cache_dir)
+            if (len(split_cache_dir) > 1 and
+                    (not os.path.exists(split_cache_dir[0]) and
+                     split_cache_dir[0] != '')):
+                if (not nilearn.EXPAND_PATH_WILDCARDS and
+                        cache_dir.startswith("~")):
+                    # Maybe the user want to enable expanded user path.
+                    error_msg = ("Given cache path base directory doesn't "
+                                 "exists, you gave '{0}'. Enabling "
+                                 "nilearn.EXPAND_PATH_WILDCARDS could solve "
+                                 "this issue.".format(split_cache_dir[0]))
+                elif self.memory.startswith("~"):
+                    # Path built on top of expanded user path doesn't exist.
+                    error_msg = ("Given cache path base directory doesn't "
+                                 "exists, you gave '{0}' which has expanded "
+                                 "as '{1}' and doesn't exist either."
+                                 .format(split_cache_dir[0],
+                                         os.path.dirname(self.memory)))
+                else:
+                    # The given cache base path doesn't exist.
+                    error_msg = ("Given cache path base directory doesn't "
+                                 "exists, you gave '{0}'."
+                                 .format(split_cache_dir[0]))
+                raise ValueError(error_msg)
```

Comment:
> So why raising an error ? If I understand the code well, my usecase the most
> common, ie memory='cache' is going to raise an error.

Nope. Because we raise an error only if the directory one step above
doesn't exist. Ie "tmp/cache" would raise an error if tmp does not
exists.